packages/flask-aserto/src/flask_aserto/aio/check.py

```diff
@@ -0,0 +1,192 @@
+    subjMapper: Optional[IdentityMapper] = None
+    policyPath: Optional[str] = ""
+    policyPathMapper: Optional[StringMapper] = None
+
+
+
+def build_resource_context_mapper(
+       opts: CheckOptions
+) -> ResourceMapper:
+
+    async def resource() -> ResourceContext:
+        objid = (
+            opts.objId
+            if opts.objId is not None
+            else ""
+        )
+        objtype = (
+            opts.objType
+            if opts.objType is not None
+            else ""
+        )
+        
+        obj = (
+            await opts.objMapper()
+            if opts.objMapper is not None
+            else Obj(id=objid, objType=objtype)
+        )
+         
+        obj.id = (
+            await opts.objIdMapper()
+            if opts.objIdMapper is not None
+            else obj.id
+        )
+
+        relation = (
+            await opts.relationMapper()
+            if opts.relationMapper is not None
+            else opts.relationName
+        )
+
+        subjType = (
+            opts.subjType
+            if opts.subjType != ""
+            else "user"
+        )
+
+        return {"relation":     relation,
+		"object_type":  obj.objType,
+		"object_id":    obj.id,
+		"subject_type": subjType}
+    
+    return resource
+
+class CheckMiddleware:
+    def __init__(
+        self,
+        *,
+        options: CheckOptions,
+        aserto_middleware: "AsertoMiddleware",
+    ):
+        self._aserto_middleware = aserto_middleware
+        
+        self._identity_provider = (
+            options.subjMapper
+            if options.subjMapper is not None
+            else aserto_middleware._identity_provider
+        )
+
+        self._resource_context_provider = build_resource_context_mapper(options)
+        self._options = options
+
+    def _with_overrides(self, **kwargs: Any) -> "CheckMiddleware":
+        return (
+            self
+            if not kwargs
+            else CheckMiddleware(
+                aserto_middleware=self._aserto_middleware,
+                options = CheckOptions(
+                    relationName=kwargs.get("relation_name", self._options.relationName),
+                    relationMapper=kwargs.get("relation_mapper", self._options.relationMapper),
+                    policyPath=kwargs.get("policy_path", self._options.policyPath),
+                    subjMapper=kwargs.get("identity_provider", self._identity_provider),
+                    objId=kwargs.get("object_id", self._options.objId),
+                    objType=kwargs.get("object_type", self._options.objType),
+                    objIdMapper=kwargs.get("object_id_mapper", self._options.objIdMapper),
+                    objMapper=kwargs.get("object_mapper", self._options.objMapper),
+                    subjType=self._options.subjType,
+                    policyPathMapper=self._options.policyPathMapper,
+                ),
+            )
+        )
+    
+    def _build_policy_context(self) -> StringMapper:
```

Comment:
The name is a bit confusing because this function doesn't return a policy context. It returns a policy path mapper.
Consider renaming it to `_build_policy_path_mapper`.